modules/phase_field/src/materials/EpsAndMAndLAnisoGauss.C

```diff
@@ -0,0 +1,931 @@
+
+    _m(declareADProperty<Real>(getParam<MaterialPropertyName>("m_name"))),
+    _m_energy(declareProperty<Real>(getParam<MaterialPropertyName>("m_energy_name"))),
+
+    _L(declareADProperty<Real>(getParam<MaterialPropertyName>("L_name"))),
+    _L_energy(declareProperty<Real>(getParam<MaterialPropertyName>("L_energy_name"))),
+
+    _sigma(declareADProperty<Real>(getParam<MaterialPropertyName>("sigma_name"))),
+    _sigmaORIUNIT(declareADProperty<Real>(getParam<MaterialPropertyName>("sigmaORIUNIT_name"))),
+
+    _qwg(declareADProperty<Real>(getParam<MaterialPropertyName>("qwg_name"))),
+
+    _qxg(declareADProperty<Real>(getParam<MaterialPropertyName>("qxg_name"))),
+
+    _qyg(declareADProperty<Real>(getParam<MaterialPropertyName>("qyg_name"))),
+
+    _qzg(declareADProperty<Real>(getParam<MaterialPropertyName>("qzg_name"))),
+
+    _epsbar(declareADProperty<Real>(getParam<MaterialPropertyName>("epsbar_name"))),
+    _epsmin2grains(declareADProperty<Real>(getParam<MaterialPropertyName>("epsmin2grains_name"))),
+
+    _TotGauss(declareADProperty<Real>(getParam<MaterialPropertyName>("TotGauss_name"))),
+
+    _sigmaBASE(getParam<Real>("sigmaBASE")),
+
+    _L_BASE(getParam<Real>("L_BASE")),
+
+    _Gaussian_Tolerance(getParam<Real>("Gaussian_Tolerance")),
+
+    _gamma(getParam<Real>("gamma")),
+
+    _lgb(getParam<Real>("lgb")),
+
+    _alphaswitch(getParam<Real>("alphaswitch")),
+
+    _betaswitch(getParam<Real>("betaswitch")),
+
+    _libnum(getParam<int>("libnum")),
+
+    _amplitudeScale(getParam<Real>("amplitudeScale")),
+
+    _sharpness(getParam<Real>("sharpness")),
+
+    _Library_file_name(getParam<FileName>("Library_file_name")),
+
+    _Quaternion_file_name(getParam<FileName>("Quaternion_file_name")),
+
+    _grain_tracker(getUserObject<GrainTrackerInterface>("grain_tracker")),
+
+    _length_scale(getParam<Real>("length_scale")),
+
+    _time_scale(getParam<Real>("time_scale")),
+
+    _JtoeV(getParam<Real>("JtoeV")),
+
+    _BoundaryNormal(getParam<Real>("BoundaryNormal")),
+
+    _op_num(coupledComponents("v")),
+
+    _vals(adCoupledValues("v")),
+
+    _grad_vals(adCoupledGradients("v"))
+
+{
+  if (_op_num == 0)
+    paramError("op_num", "op_num must be greater than 0");
+}
+
+void
+EpsAndMAndLAnisoGauss::computeQpProperties()
+{
+
+  _grain_num = _grain_tracker.getTotalFeatureCount();
+
+  const auto & op_to_grains = _grain_tracker.getVarToFeatureVector(_current_elem->id());
+
+  //************************ Read Quaternions file *************************************
+  std::ifstream File1;
+  File1.open(_Quaternion_file_name.c_str());
+
+  std::vector<double> qwR;
+  std::vector<double> qxR;
+  std::vector<double> qyR;
+  std::vector<double> qzR;
+
+  double QuatW;
+  double QuatX;
+  double QuatY;
+  double QuatZ;
+
+  while (!File1.eof())
+  {
+    File1 >> QuatW >> QuatX >> QuatY >> QuatZ;
+    qwR.push_back(QuatW);
+    qxR.push_back(QuatX);
+    qyR.push_back(QuatY);
+    qzR.push_back(QuatZ);
+  }
+
+  File1.close();
```

Comment:
Ok, 100% no. You cannot read a file at every quadrature point, this will be repeated thousands to millions of times during even small runs!

Reply:
That was fixed. See updated files.